src/Riok.Mapperly/Emit/Syntax/SyntaxFactoryHelper.cs

```diff
@@ -64,6 +64,22 @@ public static SeparatedSyntaxList<T> CommaSeparatedList<T>(params T[] nodes)
         return SeparatedList<T>(joinedNodes);
     }
 
+    public SeparatedSyntaxList<T> ConditionalCommaLineFeedSeparatedList<T>(IEnumerable<T> nodes)
+        where T : SyntaxNode
+    {
+        var nodesList = nodes.ToList();
+        var sep = Token(SyntaxKind.CommaToken);
+        if (nodesList.Count > 1)
```

Comment:
Wouldn't branching base on the nodes total length make more sense here? E.g. `if (nodesList.Sum(x => x.FullSpan.Length) > MultilineSeparatedListThreshold)`

Reply:
Hmmm, frankly I'm not sure. Having the total length as the metric might make more sense if we were also taking into account what comes before it (the type name, etc.). But if you'd feel more satisfied that way I'll change this.

Reply:
Using the total length of a line of code to decide when and where insert a NL would be the optimal solution. However, this would be very complex to implement. I think using the length of the arguments does not lead to optimal results but still better than just considering the number of arguments.